src/main.rs

```diff
@@ -253,11 +262,11 @@ async fn handle_request(
     #[cfg(feature = "expose-metrics")]
     let start = Instant::now();
 
-    let resp = match client.raw(raw_request).await.context(RequestIssue) {
+    let resp = match client.request::<Vec<u8>>(raw_request).await {
```

Comment:
This could be an if let

Reply:
It cannot due to the error being used, see https://rust-lang.github.io/rust-clippy/master/index.html#single_match_else as to why pedantic doesn't suggest this